src/tracer.js

```diff
@@ -253,6 +273,38 @@ export default class Tracer {
     }
 
     /**
+     * Reapplies sampling if the parent span isn't sampled
+     *
+     * @param parent the parent span context
+     * @param ctx the span context
+     * @param operationName the operation name for the child
+     * @param tags tags to be applied by the sampler
+     */
+    handleUpsampling(parent: SpanContext, ctx: SpanContext, operationName: string, tags: any) {
+        if (!this._upsampling.enabled) {
+            return
+        }
+
+        // Run upsampler on every span
+        if (this._upsampling.allowOnInnerSpans) {
+            if (this._sampler.isSampled(operationName, tags)) {
```

Comment:
@ys is this really what we want? should allowOnInnerSpans allow you to have different sampling decisions for each span in the process? I don't see the use case for this, if the parent span in the process is upsampled, all child spans should respect it and be upsampled. Don't know why we'd want to flip the coin for each span.